pkg/controller/migration/convert/convert.go

```diff
@@ -0,0 +1,55 @@
+// package convert reads config from existing Calico installations that are not
+// managed by Operator, and generates Operator Config that can be used
+// to configure a similar cluster.
+package convert
+
+import (
+	"context"
+	"fmt"
+
+	operatorv1 "github.com/tigera/operator/pkg/apis/operator/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+)
+
+var log = logf.Log.WithName("migration_convert")
+
+var ctx = context.Background()
+
+// Convert updates an Installation resource based on an existing Calico install (i.e.
+// one that is not managed by operator). If the existing installation cannot be represented by an Installation
+// resource, an ErrIncompatibleCluster is returned.
+func Convert(ctx context.Context, client client.Client, install *operatorv1.Installation) error {
+	comps, err := getComponents(ctx, client)
+	if err != nil {
+		if kerrors.IsNotFound(err) {
+			log.Error(err, "no existing install found: %v", err)
```

Comment:
Unclear if this should be demoted to an info log, or if we should be returning an error here instead of nil.

Also, I think this error message includes `err` twice? Probably can remove the `"%v", err` from the end.